go-controller/pkg/ovn/secondary_layer2_network_controller.go

```diff
@@ -556,6 +560,74 @@ func (oc *SecondaryLayer2NetworkController) addUpdateLocalNodeEvent(node *corev1
 	return err
 }
 
+func (oc *SecondaryLayer2NetworkController) addUpdateRemoteNodeEvent(node *corev1.Node, syncZoneIC bool) error {
+	var errs []error
+
+	if util.IsNetworkSegmentationSupportEnabled() && oc.IsPrimaryNetwork() {
+		if syncZoneIC && config.OVNKubernetesFeature.EnableInterconnect {
+			if err := oc.addPortForRemoteNodeGR(node); err != nil {
+				err = fmt.Errorf("failed to add the remote zone node [%s] to the zone interconnect handler, err : %v", node.Name, err)
+				errs = append(errs, err)
+				oc.syncZoneICFailed.Store(node.Name, true)
+			} else {
+				oc.syncZoneICFailed.Delete(node.Name)
+			}
+		}
+	}
+
+	errs = append(errs, oc.BaseSecondaryLayer2NetworkController.addUpdateRemoteNodeEvent(node))
+
+	err := utilerrors.Join(errs...)
+	if err != nil {
+		oc.recordNodeErrorEvent(node, err)
+	}
+	return err
+}
+
+func (oc *SecondaryLayer2NetworkController) addPortForRemoteNodeGR(node *corev1.Node) error {
+	nodeJoinSubnetIPs, err := util.ParseNodeGatewayRouterJoinAddrs(node, oc.GetNetworkName())
+	if err != nil || len(nodeJoinSubnetIPs) == 0 {
+		return fmt.Errorf("failed to get the node %s join subnet IPs : %w", node.Name, err)
+	}
+
+	remoteGRPortMac := util.IPAddrToHWAddr(nodeJoinSubnetIPs[0].IP)
+	var remoteGRPortNetworks []string
+	for _, ip := range nodeJoinSubnetIPs {
+		remoteGRPortNetworks = append(remoteGRPortNetworks, ip.String())
+	}
+
+	remotePortAddr := remoteGRPortMac.String()
+	for _, tsNetwork := range remoteGRPortNetworks {
+		remotePortAddr = remotePortAddr + " " + tsNetwork
+	}
```

Comment:
Is it worth it to use strings.Join to remove the loop here ? 
```golang
remotePortAddr :=  remoteGRPortMac.String() +  " " + strings.Join(remoteGRPortNetworks, " ")
```
Don't know if this improve the readability.

Reply:
sure  I don't mind doing that